packages/app/src/vis-packs/core/complex/ComplexLineVisContainer.tsx

```diff
@@ -20,19 +21,19 @@ function ComplexLineVisContainer(props: VisContainerProps) {
   const { shape: dims } = entity;
   const [dimMapping, setDimMapping] = useDimMappingState(dims, 1);
 
+  const autoScale = useLineConfig((state) => state.autoScale);
+
   return (
     <>
       <DimensionMapper
         rawDims={dims}
         mapperState={dimMapping}
         onChange={setDimMapping}
       />
-      <VisBoundary
-        resetKey={dimMapping}
-        loadingMessage="Loading entire dataset"
-      >
+      <VisBoundary resetKey={dimMapping} loadingMessage="Loading current slice">
```

Comment:
The loading message depends on `autoScale`. Might be worth adding a boolean prop to move the logic of the message inside `VisBoundary`.

Reply:
Well, I was thinking this would apply to `HeatmapVisContainer` and the like as well, but maybe you're right, it's a bit more complex than that since for NeXus visualizations, we load more than one dataset...

Reply:
ALl right, I will but a ternary on both `ComplexLineVisContainer` and `LineVisContainer` and call it a day